### PR TITLE
new package: py-numcodecs

### PR DESCRIPTION
**Pull request by Sinan81**

### CODE REVIEW

**Comment by Sinan81** [comment]
a dependency of #14326

File: var/spack/repos/builtin/packages/py-numcodecs/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "https://github.com/zarr-developers/numcodecs"
+    url      = "https://pypi.io/packages/source/n/numcodecs/numcodecs-0.6.4.tar.gz"
+
+    version('0.6.4', sha256='ef4843d5db4d074e607e9b85156835c10d006afc10e175bda62ff5412fca6e4d')
+
+    depends_on('python@2.7:2.8,3.5:',       type=('build', 'run'))
+    depends_on('py-setuptools@18.0:',       type='build')
+    depends_on('py-setuptools-scm@1.5.4:',  type=('build', 'run'))
+    depends_on('py-numpy@1.7:',             type=('build', 'run'))
+    depends_on('py-msgpack',                type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numcodecs/package.py]: I would make this optional using a variant.

File: var/spack/repos/builtin/packages/py-numcodecs/package.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyNumcodecs(PythonPackage):
+    """FIXME: Put a proper description of your package here."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numcodecs/package.py]: Needs a description.

File: var/spack/repos/builtin/packages/py-numcodecs/package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    homepage = "https://github.com/zarr-developers/numcodecs"
+    url      = "https://pypi.io/packages/source/n/numcodecs/numcodecs-0.6.4.tar.gz"
+
+    version('0.6.4', sha256='ef4843d5db4d074e607e9b85156835c10d006afc10e175bda62ff5412fca6e4d')
+
+    depends_on('python@2.7:2.8,3.5:',       type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numcodecs/package.py]: I would remove this line.

**Comment by Sinan81** [comment]
Thanks Adam!

**Comment by adamjstewart** [comment]
Needs to be rebased now that `py-msgpack` has been merged.

**Pull request merged**